### PR TITLE
Grouped details list/column reorder

### LINKED ISSUE #25983: [Bug]: Reordering grouped DetailsList columns throws an error
### Library

React / v8 (@fluentui/react)

### System Info

```shell
n/a
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codepen.io/seanms/pen/vYrqpZv

## Actual Behavior

In a grouped `DetailsList` with drag and drop enabled, dragging the right-most column to the left throws an error.

https://user-images.githubusercontent.com/93940821/207469258-7150e461-6ff1-4b32-9aaa-7f5ae9da989f.mp4

## Expected Behavior

I should be able to drag and drop the columns

### Logs

```shell
Uncaught TypeError: Cannot read properties of undefined (reading 'minWidth')
```

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by spmonahan**
## Previous Behavior

When dragging a column in a grouped DetailsList it is possible to throw a runtime error that breaks rendering. This is because column offsets were not consistently computed.

## New Behavior

Grouped DetailsList no longer throws a runtime error when dragging and dropping columns. Column offset calculations are moved into a single function so they can be consistently applied.

## Related Issue(s)

* Fixes #25983

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/15328c336f754535450e7190802b9ea4b948f5e6'>15328c336f754535450e7190802b9ea4b948f5e6</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the 34 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26015&commitId=d7fbc71664754e0f467d6f3316ab79427e5fcdea&env=prod)

✅ There was 2 screenshots added, 24 screenshots removed, 1011 screenshots unchanged, 0 screenshots with different dimensions and 8 screenshots with visible difference.


  <details><summary><b>unknown</b> 34 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|DetailsHeader.Root.default.chromium.png|260337| Changed|
|DetailsList.Checkbox Visible Always - RTL.default.chromium.png|417738| Changed|
|DetailsList.Checkbox Visible Always.default.chromium.png|392818| Changed|
|DetailsList.Compact.default.chromium.png|360573| Changed|
|DetailsList.Root.default.chromium.png|393080| Changed|
|DetailsList.Single Selection Mode.default.chromium.png|393080| Changed|
|ScrollablePane Details List.ScrollablePane Details List with sticky header & footer.default.chromium.png|528914| Changed|
|Sticky breadcrumb and sticky details list header.ScrollablePane Sticky Breadcrumb Details List.default.chromium.png|760356| Changed|
|Pivot - Overflow.Tabs - RTL.Narrow - Last tab selected.chromium.png|0| Added|
|Pivot - Overflow.Tabs - RTL.Narrow - Overflow menu.chromium.png|0| Added|
|DetailsHeader.Root.borderWhileDragging.chromium.png|0| Removed|
|DetailsHeader.Root.dropHint.chromium.png|0| Removed|
|DetailsHeader.Root.hoverDraggable.chromium.png|0| Removed|
|DetailsHeader.Root.hoverFrozenFirst.chromium.png|0| Removed|
|DetailsHeader.Root.hoverFrozenLast.chromium.png|0| Removed|
|DetailsList.Checkbox Visible Always - RTL.click.chromium.png|0| Removed|
|DetailsList.Checkbox Visible Always - RTL.hover.chromium.png|0| Removed|
|DetailsList.Checkbox Visible Always.click.chromium.png|0| Removed|
|DetailsList.Checkbox Visible Always.hover.chromium.png|0| Removed|
|DetailsList.Compact.click.chromium.png|0| Removed|
|DetailsList.Compact.hover.chromium.png|0| Removed|
|DetailsList.Root.click.chromium.png|0| Removed|
|DetailsList.Root.hover.chromium.png|0| Removed|
|DetailsList.Single Selection Mode.click.chromium.png|0| Removed|
|DetailsList.Single Selection Mode.hover.chromium.png|0| Removed|
|ScrollablePane Details List.ScrollablePane Details List with sticky header & footer.scroll down by a small amount so that the first row is still visible.chromium.png|0| Removed|
|ScrollablePane Details List.ScrollablePane Details List with sticky header & footer.scroll down to the bottom.chromium.png|0| Removed|
|ScrollablePane Details List.ScrollablePane Details List with sticky header & footer.scroll up to the top.chromium.png|0| Removed|
|Sticky breadcrumb and sticky details list header.ScrollablePane Sticky Breadcrumb Details List.scroll back to the top when horizontal scroll is non-zero.chromium.png|0| Removed|
|Sticky breadcrumb and sticky details list header.ScrollablePane Sticky Breadcrumb Details List.scroll down by a small amount so that the first row is still visible and the header becomes sticky.chromium.png|0| Removed|
|Sticky breadcrumb and sticky details list header.ScrollablePane Sticky Breadcrumb Details List.scroll down to the bottom.chromium.png|0| Removed|
|Sticky breadcrumb and sticky details list header.ScrollablePane Sticky Breadcrumb Details List.scroll down when horizontal scroll is non-zero.chromium.png|0| Removed|
|Sticky breadcrumb and sticky details list header.ScrollablePane Sticky Breadcrumb Details List.scroll right.chromium.png|0| Removed|
|Sticky breadcrumb and sticky details list header.ScrollablePane Sticky Breadcrumb Details List.scroll up to the top.chromium.png|0| Removed|

</details>


<div id="vrtCommentfluentuiv8"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/BaseButton-mount_base.html">1189</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/BaseButton-mount.html">1195</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Breadcrumb-mount_base.html">2815</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Breadcrumb-mount.html">2794</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Checkbox-mount_base.html">2590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Checkbox-mount.html">2644</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/CheckboxBase-mount_base.html">2329</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/CheckboxBase-mount.html">2351</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ChoiceGroup-mount_base.html">4330</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ChoiceGroup-mount.html">4249</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ComboBox-mount_base.html">1214</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ComboBox-mount.html">1179</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/CommandBar-mount_base.html">9354</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/CommandBar-mount.html">9315</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ContextualMenu-mount_base.html">10415</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ContextualMenu-mount.html">10245</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DefaultButton-mount_base.html">1360</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DefaultButton-mount.html">1383</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DetailsRow-mount_base.html">3366</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DetailsRow-mount.html">3340</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DetailsRowFast-mount_base.html">3342</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DetailsRowFast-mount.html">3387</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DetailsRowNoStyles-mount_base.html">3231</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DetailsRowNoStyles-mount.html">3234</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Dialog-mount_base.html">2919</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Dialog-mount.html">2949</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DocumentCardTitle-mount_base.html">566</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/DocumentCardTitle-mount.html">571</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Dropdown-mount_base.html">3168</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Dropdown-mount.html">3159</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/FocusTrapZone-mount_base.html">1949</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/FocusTrapZone-mount.html">1963</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/FocusZone-mount_base.html">1938</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/FocusZone-mount.html">1906</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedList-mount_base.html">1808</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedList-mount.html">2061</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedList-virtual-rerender_base.html">1088</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedList-virtual-rerender.html">1084</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">1615</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedList-virtual-rerender-with-unmount.html">1605</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedListV2-mount_base.html">559</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedListV2-mount.html">566</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedListV2-virtual-rerender_base.html">539</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedListV2-virtual-rerender.html">541</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">568</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">554</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/IconButton-mount_base.html">1748</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/IconButton-mount.html">1780</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Label-mount_base.html">789</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Label-mount.html">746</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Layer-mount_base.html">4161</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Layer-mount.html">4179</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Link-mount_base.html">849</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Link-mount.html">860</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/MenuButton-mount_base.html">1587</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/MenuButton-mount.html">1599</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/MessageBar-mount_base.html">2361</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/MessageBar-mount.html">2314</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Nav-mount_base.html">3048</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Nav-mount.html">3044</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/OverflowSet-mount_base.html">1418</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/OverflowSet-mount.html">1397</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Panel-mount_base.html">2490</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Panel-mount.html">2444</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Persona-mount_base.html">1285</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Persona-mount.html">1302</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Pivot-mount_base.html">1512</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Pivot-mount.html">1510</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/PrimaryButton-mount_base.html">1486</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/PrimaryButton-mount.html">1491</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Rating-mount_base.html">6907</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Rating-mount.html">6947</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/SearchBox-mount_base.html">1494</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/SearchBox-mount.html">1499</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Shimmer-mount_base.html">2853</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Shimmer-mount.html">2857</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Slider-mount_base.html">2093</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Slider-mount.html">2109</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/SpinButton-mount_base.html">4274</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/SpinButton-mount.html">4459</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Spinner-mount_base.html">825</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Spinner-mount.html">831</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/SplitButton-mount_base.html">2831</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/SplitButton-mount.html">2812</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Stack-mount_base.html">865</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Stack-mount.html">848</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/StackWithIntrinsicChildren-mount_base.html">2272</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/StackWithIntrinsicChildren-mount.html">2256</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/StackWithTextChildren-mount_base.html">4932</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/StackWithTextChildren-mount.html">4975</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/SwatchColorPicker-mount_base.html">9451</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/SwatchColorPicker-mount.html">9382</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/TagPicker-mount_base.html">2331</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/TagPicker-mount.html">2361</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/TeachingBubble-mount_base.html">73844</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/TeachingBubble-mount.html">73890</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Text-mount_base.html">819</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Text-mount.html">835</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/TextField-mount_base.html">1540</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/TextField-mount.html">1576</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ThemeProvider-mount_base.html">1447</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ThemeProvider-mount.html">1444</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ThemeProvider-virtual-rerender_base.html">1135</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ThemeProvider-virtual-rerender.html">1140</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2005</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1996</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Toggle-mount_base.html">1138</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/Toggle-mount.html">1151</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/buttonNative-mount_base.html">539</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26015/perf-test/buttonNative-mount.html">540</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/DetailsList/DetailsHeader.base.tsx [hunk 1]
```diff
@@ -425,6 +425,21 @@ export class DetailsHeaderBase
     this._draggedColumnIndex = -1;
   };
 
+  private _computeColumnIndexOffset = (showCheckbox: boolean) => {
+    const hasGroupExpander = this.props.groupNestingDepth! > 0;
```
Comment (behowell) [inline packages/react/src/components/DetailsList/DetailsHeader.base.tsx:429]: Avoid non-null assertions `!`. Could this instead be a truthy check? 
```suggestion
    const hasGroupExpander = !!this.props.groupNestingDepth;
```

Or if checking for non-negative values is important:
```suggestion
    const hasGroupExpander = !!this.props.groupNestingDepth && this.props.groupNestingDepth > 0;
```

**Review by behowell — APPROVED** [review]

**Pull request merged**